client/src/featureform/resources.py

```diff
@@ -984,22 +987,41 @@ def to_key(self) -> Tuple[str, str, str]:
 @typechecked
 @dataclass
 class PrimaryData:
-    location: Location
```

Comment:
Just for my knowledge, was this change just to document it can be none?

Reply:
Yes, in the case of `get_source`, we don't have the information to determine which kind of location it is, so to satisfy the types, I allowed it to be `None`